src/Config.php

```diff
@@ -1658,4 +1640,80 @@ public function printConfigData($data)
     }//end printConfigData()
 
 
+    /**
+     * Parse supplied string into a list of sniff codes.
+     *
+     * @param string $input    Comma-separated string of sniff codes.
+     * @param string $argument The name of the argument which is being processed.
+     *
+     * @return string[]
+     * @throws DeepExitException
+     */
+    private function parseSniffCodes($input, $argument)
+    {
+        $errors = [];
+        $sniffs = [];
+
+        $possibleSniffs = explode(',', $input);
+        $possibleSniffs = array_unique($possibleSniffs);
+
+        foreach ($possibleSniffs as $sniff) {
+            $sniff = trim($sniff);
+
+            if ($sniff === '') {
+                // Empty values can be safely ignored.
+                continue;
+            }
+
+            if (preg_match('{[^A-Za-z0-9.]}', $sniff) === 1) {
+                $errors[] = 'Unsupported character detected: '.$sniff;
+                continue;
+            }
+
+            $partCount = substr_count($sniff, '.');
+            if ($partCount === 2) {
+                // Correct number of parts.
+                $sniffs[] = $sniff;
+                continue;
+            }
+
+            if ($partCount === 0) {
+                $errors[] = 'Standard codes are not supported: '.$sniff;
+            } else if ($partCount === 1) {
+                $errors[] = 'Category codes are not supported: '.$sniff;
+            } else if ($partCount === 3) {
+                $errors[] = 'Message codes are not supported: '.$sniff;
+            } else {
+                $errors[] = 'Too many parts: '.$sniff;
+            }
+
+            if ($partCount > 2) {
+                $parts    = explode('.', $sniff, 4);
+                $sniffs[] = $parts[0].'.'.$parts[1].'.'.$parts[2];
+            }
+        }//end foreach
+
+        if ($errors !== []) {
+            $error  = 'ERROR: The --'.$argument.' option only supports sniff codes.'.PHP_EOL;
+            $error .= 'Sniff codes are in the form "Standard.Category.Sniff"'.PHP_EOL;
+            $error .= PHP_EOL;
+            $error .= 'The following problems were detected:'.PHP_EOL;
+            $error .= '* '.implode(PHP_EOL.'* ', $errors).PHP_EOL;
+
+            if ($sniffs !== []) {
+                $sniffs = array_unique($sniffs);
```

Comment:
I understand why the `array_unique()` is here, but that does make the case-sensitive issue (more) relevant.

```
Perhaps try --sniffs="Generic.Arrays.DisallowShortArray,generic.arrays.disallowshortarray,Generic.PHP.BacktickOperator,generic.php.backtickoperator
```